File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/logic/function/process/condition/impl/EntityOwnerCondition.java
package io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.impl;

import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.Process;
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.Condition;
import org.bukkit.entity.Entity;
import org.bukkit.entity.LivingEntity;
import org.jetbrains.annotations.NotNull;
import org.spongepowered.configurate.CommentedConfigurationNode;

public class EntityOwnerCondition extends Condition {

    public EntityOwnerCondition(
        Process parentProcess,
        @NotNull CommentedConfigurationNode conditionNode
    ) {
        super(parentProcess, conditionNode);
        //TODO parse modal list
    }

    @Override
    public boolean applies(Context context) {
        final Entity entity = context.getEntity();
        if(entity == null) {
            throw new IllegalStateException("Missing entity context");
        }
        if(!(entity instanceof LivingEntity lent)) {
            throw new IllegalStateException("Entity context is not a LivingEntity type");
        }

        //TODO check integrations

        return true;
    }
}

File: levelledmobs-plugin-bukkit/src/main/kotlin/io/github/arcaneplugins/levelledmobs/bukkit/logic/function/process/condition/impl/EntityOwnerCondition.kt
package io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.impl

import io.github.arcaneplugins.levelledmobs.bukkit.logic.context.Context
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.Process
import io.github.arcaneplugins.levelledmobs.bukkit.logic.function.process.condition.Condition
import org.spongepowered.configurate.CommentedConfigurationNode

class EntityOwnerCondition(
    process: Process,
    node: CommentedConfigurationNode
): Condition(process, node) {
    override fun applies(context: Context): Boolean {

        requireNotNull(context.livingEntity) { "LivingEntity context must not be null" }

        //TODO check integrations


        //TODO check integrations
        return true
    }
}